apps/server/src/modules/files-storage/uc/files-storage.uc.ts

```diff
@@ -71,27 +68,36 @@ export class FilesStorageUC {
 
 	// upload
 	public async upload(userId: EntityId, params: FileRecordParams, req: Request): Promise<FileRecord> {
-		await this.checkPermission(
-			userId,
-			params.parentType,
-			params.parentId,
-			FileStorageAuthorizationContext.create(params.storageLocation)
-		);
+		await this.checkPermission(userId, params.parentType, params.parentId, FileStorageAuthorizationContext.create);
 
-		await this.checkStorageLocation(params.storageLocation, params.storageLocationId);
+		await this.checkStorageLocation(userId, params.storageLocation, params.storageLocationId);
```

Comment:
await Promise.all([
     this.checkPermission(userId, params.parentType, params.parentId, FileStorageAuthorizationContext.create),
     this.checkStorageLocation(userId, params.storageLocation, params.storageLocationId)
]);